File: src/main/java/com/freestudy/api/infra/slack/SlackMessageEvent.java
package com.freestudy.api.infra.slack;

import lombok.Getter;
import org.springframework.context.ApplicationEvent;

public class SlackMessageEvent extends ApplicationEvent {

  @Getter
  private String message;

  /**
   * Create a new ApplicationEvent.
   *
   * @param source the object on which the event initially occurred (never {@code null})
   */
  public SlackMessageEvent(Object source, String message) {
    super(source);
    this.message = message;
  }
}

File: src/main/kotlin/social/alone/server/slack/SlackMessageEvent.kt
package social.alone.server.slack

import lombok.Getter
import org.springframework.context.ApplicationEvent

class SlackMessageEvent
/**
 * Create a new ApplicationEvent.
 *
 * @param source the object on which the event initially occurred (never `null`)
 */
(source: Any, @field:Getter
val message: String) : ApplicationEvent(source)
